metrics/src/metric_types.rs

```diff
@@ -124,3 +126,40 @@ impl Gauge {
         }
     }
 }
+
+use once_cell::sync::OnceCell;
+use parking_lot::RwLock;
+
+const QUEUE_CAPACITY: usize = 256;
+
+/// A histogram backed by a circular queue.
+pub struct CircularHistogram(OnceCell<RwLock<CircularQueue<f64>>>);
```

Comment:
I'm assuming the `OnceCell` is here so that the RPC-related objects can be created with a `const fn`? It's probably cheap enough, but I'm wondering if this could be made simpler :thinking:; otherwise you could add a non-doc comment here on why `OnceCell` is used.

Reply:
Yes—I'm open to suggestions, I've added the doc comment in the meantime.